### PR TITLE
OCL: [UMat::convertTo]It is unnecessary to use fma() if no scaling.

### PR DESCRIPTION
**Pull request by wangyan42164**
Signed-off-by: Yan Wang <yan.wang@linux.intel.com>

### CODE REVIEW

**Comment by alalek** [comment]
Looks good!

BTW, In this simplified line:
```
dst[0] = convertToDT(convertToWT(src[0]));
```
I believe we can eliminate conversion to "work type" (`convertToWT`). What do you think, will it work?

**Comment by vpisarev** [comment]
indeed, 2 conversions seem unnecessary, but the compiler should optimize it out, so it's not that important

**Comment by paroj** [comment]
does the whole function have to be copied to essentially only change one line?

**Comment by alalek** [comment]
Arguments were changed also. But it is the right note about code duplication.

**Comment by wangyan42164** [comment]
@alalek @vpisarev @paroj Thanks for your comments. I have refined this PR and submitted again. Please help review it.

**Comment by wangyan42164** [comment]
@alalek 
For Android build machine:
It seems that this PR isn't the reason of warning.
For Mac build machine:
I am not sure the following failure which comes from removing convertToWT and convertToDT. If true, I could roll back  convertToWT and convertToDT. I haven't Mac machine to verify it on hand. Could you please help me?
Thanks.

[----------] 2 tests from Core_HAL
[ RUN      ] Core_HAL.mathfuncs
exp (N=100, f32): hal time=0.50usec, ocv time=0.62usec
exp (N=100, f64): hal time=0.68usec, ocv time=0.78usec
log (N=100, f32): hal time=0.44usec, ocv time=0.67usec
log (N=100, f64): hal time=0.90usec, ocv time=1.04usec
sqrt (N=100, f32): hal time=0.18usec, ocv time=0.40usec
sqrt (N=100, f64): hal time=0.44usec, ocv time=0.65usec
[       OK ] Core_HAL.mathfuncs (127 ms)
[ RUN      ] Core_HAL.mat_decomp
LU (4 x 4, f32): hal time=0.32usec, ocv time=0.81usec
LU (4 x 4, f64): hal time=0.36usec, ocv time=0.84usec
LU (6 x 6, f32): hal time=0.61usec, ocv time=1.14usec
LU (6 x 6, f64): hal time=0.64usec, ocv time=1.15usec
/build/precommit_macosx/opencv/modules/core/test/test_hal_core.cpp:186: Failure
Expected: (norm(x, x0, NORM_INF | NORM_RELATIVE)) <= (eps), actual: 1.78846 vs 1e-05
LU (15 x 15, f32): hal time=5.08usec, ocv time=4.49usec
LU (15 x 15, f64): hal time=4.80usec, ocv time=5.70usec
Cholesky (4 x 4, f32): hal time=0.24usec, ocv time=0.72usec
Cholesky (4 x 4, f64): hal time=0.20usec, ocv time=0.69usec
Cholesky (6 x 6, f32): hal time=0.44usec, ocv time=0.97usec
Cholesky (6 x 6, f64): hal time=0.34usec, ocv time=0.85usec
Cholesky (15 x 15, f32): hal time=3.27usec, ocv time=3.90usec
Cholesky (15 x 15, f64): hal time=2.62usec, ocv time=3.56usec
[  FAILED  ] Core_HAL.mat_decomp (1 ms)
[----------] 2 tests from Core_HAL (128 ms total)

**Comment by wangyan42164** [comment]
@alalek Now build machines have only just Android warnings. Thanks.

File: modules/core/src/opencl/convert.cl [hunk 1]
```diff
@@ -68,7 +71,11 @@ __kernel void convertTo(__global const uchar * srcptr, int src_step, int src_off
             __global const srcT * src = (__global const srcT *)(srcptr + src_index);
             __global dstT * dst = (__global dstT *)(dstptr + dst_index);
 
+#ifdef NO_SCALE
+            dst[0] = src[0];
```
Comment (alalek) [inline modules/core/src/opencl/convert.cl]: `convertToDT` can include rounding policy (_rte, _sat) that may differ from the default type cast:
https://github.com/Itseez/opencv/blob/c0b544af705eb058642cd8387abf3d9536eaa32a/modules/core/src/ocl.cpp#L5585-L5603

Comment (vpisarev) [inline modules/core/src/opencl/convert.cl]: agree, convertToDT needs to be retained

**Comment by wangyan42164** [comment]
@alalek @vpisarev I have add convertToDT back. Please help check it. Thanks.

**Comment by wangyan42164** [comment]
@alalek If no more issues, could you please merge it? Thanks.

**Comment by alalek** [comment]
:+1:

**Pull request merged**